src/status/libstatus/accounts.nim

```diff
@@ -225,7 +225,7 @@ proc storeDerivedWallet*(account: GeneratedAccount, password: string, walletInde
   let hashedPassword = hashPassword(password)
   let multiAccount = %* {
     "accountID": account.id,
-    "paths": ["m/" & $walletIndex],
+    "paths": ["m/44'/60'/0'/0/" & $walletIndex],
```

Comment:
Okay, few things to note here:

This function `storeDerivedWallet` is called as part of `saveAccount` here:

https://github.com/status-im/nim-status-client/blob/68aafaba6fdee1521bdf6075f35b1e6b75a2f2a6/src/status/libstatus/accounts.nim#L241-L242

Notice how it's only called if the given `GeneratedAccount` is considered a **derived account**. This has confused me a lot. The account passed in this particular scenario is **not** a derived account, but the "master" account we get from importing the mnemonic here: 

https://github.com/status-im/nim-status-client/blob/68aafaba6fdee1521bdf6075f35b1e6b75a2f2a6/src/status/wallet.nim#L189

So **actually**, `isADerivedAAccount` should be set to `false` here: 

https://github.com/status-im/nim-status-client/blob/68aafaba6fdee1521bdf6075f35b1e6b75a2f2a6/src/status/wallet.nim#L190

However, that would mean, we would not call `storeDerivedWallet` at all, which doesn't seem to be correct either, because we have to store it so it can be used for transactions...

So after having tons of back and forth discussions with @gravityblast we came to the conclusion that the path here needs to be changed. This makes everything work nicely, but I'm not sure if this function was/is supposed to be used in other scenarios where the wallet/account should be stored in a different path.

Also, I assume it was set to `m/0` for a reason.

I checked the code base and this function is only used in the case when a user imports an account inside of the wallet. So it should be fine for now.

Please let me know if I'm missing anything here @jrainville @richard-ramos

Reply:
I think it should be fine. I was personally a bit lost when I implemented that in the first place, so  think I misunderstood what a derived account is, hence why the flag `isADerivedAAccount` is wrong